modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1949 @@
+         *
+         * For example
+         * ```
+         * Quatd q1{1,2,3,4};
+         * cout << log(q1) << endl;
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> log(const Quat<T> &q, bool assumeUnit);
+
+        /**
+         * @brief return the value of logarithm function
+         *  \f[\ln(q) = \ln||q|| + \frac{\boldsymbol{v}}{||\boldsymbol{v}||}\arccos\frac{w}{||q||}\f]
+         * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatd q(1,2,3,4);
+         * q.log();
+         *
+         * Quatd q1(1,2,3,4);
+         * q1.normalize().log(true);
+         * ```
+         */
+        Quat<_Tp> log(bool assumeUnit=false) const;
+
+        /**
+         * @brief return the value of power function with index \f$x\f$
+         * \f[q^x = ||q||(cos(x\theta) + \boldsymbol{u}sin(x\theta)))\f]
+         * @param q a quaternion
+         * @param x index of exponentiation
+         * @param assumeUnit if true, quaternion q should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * quatd q(1,2,3,4);
+         * power(q, 2);
+         *
+         * double angle = CV_PI;
+         * Vec3d axis{0, 0, 1};
+         * Quatd q1(angle, axis); //generate a unit quat by axis and angle
+         * power(q1, 2, true);
+         * ```
+         */
+        template <typename T, typename _T>
+        friend Quat<T> power(const Quat<T> &q, _T x, bool assumeUnit);
+
+        /**
+         * @brief return the value of power function with index \f$x\f$
+         * \f[q^x = ||q||(\cos(x\theta) + \boldsymbol{u}\sin(x\theta)))\f]
+         * @param x index of exponentiation
+         * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * quatd q(1,2,3,4);
+         * q.power(2);
+         *
+         * double angle = CV_PI;
+         * Vec3d axis{0, 0, 1};
+         * Quatd q1(angle, axis); //generate a unit quat by axis and angle
+         * q1.power(2, true); //true means q is a unit quaternion
+         * ```
+         */
+        template <typename _T>
+        Quat<_Tp> power(_T x, bool assumeUnit=false) const;
+
+        /**
+         * @brief return \f$\sqrt{q}\f$
+         * @param q a quaternion
+         * @param assumeUnit if true, quaternion q should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatf q(1,2,3,4);
+         * sqrt(q);
+         *
+         * q = {1,0,0,0};
+         * sqrt(q, true); //true means q is a unit quaternion
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> sqrt(const Quat<T> &q, bool assumeUnit);
```

Comment:
It can be implemented without friendship, because there is public method  `sqrt`.